lib/compress.js

```diff
@@ -1053,7 +1053,7 @@ merge(Compressor.prototype, {
             var val = {};
             for (var i = this.properties.length; --i >= 0;) {
                 var prop = this.properties[i];
-                if (prop instanceof AST_ObjectKeyVal) {
+                if (typeof prop.key === "string") {
```

Comment:
On both master and harmony `ev()` always to be evaluated for `prop.value` in case it throws.